### ISSUE TITLE
 Support for drawing with the mouse on a chartCanvas

### ISSUE THREAD

**Issue by Serg65s** [d3fc/d3fc#1667]
Is there a way to correctly draw on the canvas using the mouse so that the drawing supports mouse events and does not interfere with the drawing of the graph, and vice versa, so that the drawing of the graph does not interfere with the drawing?

 It would be desirable for the drawing to respond correctly to the scale of the chart.

Are there any possibilities for arbitrary rendering, for example, via Path SVG with mouse support?

**Comment by chrisprice** [comment]
There's no built in support but I don't see why it couldn't be done. You'll need to decorate in some event handlers to capture the mouse interactions and a custom component to render the captured drawing.

**Comment by Serg65s** [comment]
Thank you so much for your reply. I don't get the mutual "isolation" of the rendering. For some reason, all the time the built-in rendering affects the layers and vice versa. I'm probably doing something wrong.:(I would be very grateful for a minimal working example so that I understand the principle.

**Comment by chrisprice** [comment]
You're right to be confused about the isolation, there isn't any 🙂 Specifically for the canvas components, a render of the chart involves clearing the canvas and then rendering each component onto the canvas in turn. 
 
We don't have a ready made example but if I were creating one I'd use a GL series for the data. Then add a canvas multi series to force the chart to render the canvas but not render anything into it. Finally, either grab an external reference to that canvas agnostic of d3fc or add a custom component to the multi to render it as part of the chart lifecycle.

Unfortunately I don't have time to create an example myself at the moment.
